src/components/Staff/Checkin/QR.tsx

```diff
@@ -0,0 +1,99 @@
+import type { CheckInDataDTO, UserShowedData } from "@/types/staff";
+import { Result } from "@zxing/library";
+import { motion } from "framer-motion";
+import { useEffect, useState } from "react";
+import { QrReader } from "react-qr-reader";
+import Failed from "./Failed";
+import User from "./User";
+const QR = () => {
+  const [isScanned, setIsScanned] = useState<boolean>(false);
+  const [showModal, setShowModal] = useState<boolean>(false);
+  const [data, setData] = useState<string | null>(null);
+  const [scanndedData, setScannedData] = useState<UserShowedData | null>(null);
+  const [textValue, setTextValue] = useState<string>("");
+  useEffect(() => {
+    if (data && !isScanned) {
+      setIsScanned(true);
+      checkIn(data);
+      setData(null);
+    }
+  }, [data]);
+  const handleScanResult = (token: Result, error: Error) => {
+    if (token) {
+      setData(token.getText);
+      setShowModal(true);
+    }
+    if (error) {
+      console.info(error);
+    }
+  };
+  const checkIn = async (token: string) => {
+    try {
+      // check-in qr code
+      const res = await fetch(
+        import.meta.env.PUBLIC_API_BASE_URL + "/staff/checkin/" + token,
+        { method: "POST" }
+      );
+      setShowModal(true);
+      if (res.ok) {
+        const user = res.json() as unknown as CheckInDataDTO;
+        setScannedData({ ...user.user, id: token });
+      } else {
+        setScannedData(null);
+      }
+    } catch {
+      console.log("Error! There was an error.");
```

Comment:
Please use `console.*` only for testing in your local environment, and refrain from pushing it into the codebase.